### PR TITLE
[IMP] Master clean/add sms phone widget

### PR DESCRIPTION
**Pull request by std-odoo**
[REF] sms: clean phone widget

The sms extention for the phone widget wasn't
compliant with JS template and needed a refactor.

[IMP] sms: add SMS icon on fields for contact

In the partner form, in the page "Contacts and Addresses",
Add next to the phone and mobile fields
the sms option to send directly message from
the kanban view and also from the small form when we click on
a card.

task-2086735

### CODE REVIEW

File: addons/sms/static/src/js/fields_phone_widget.js [hunk 1]
```diff
@@ -41,8 +44,8 @@ Phone.include({
             views: [[false, 'form']],
             context: context,
         }, {
-        on_close: function () {
-            self.trigger_up('reload');
+        on_close: () => {
```
Comment (tde-banana-odoo) [inline addons/sms/static/src/js/fields_phone_widget.js]: Please no .

File: addons/sms/static/src/js/fields_phone_widget.js [hunk 2]
```diff
@@ -53,20 +56,12 @@ Phone.include({
      * @private
      */
     _renderReadonly: function () {
-        var def = this._super.apply(this, arguments);
+        //this.$el.html(QWeb.render('field_phone_widget_readonly', {'widget': this}));
```
Comment (tde-banana-odoo) [inline addons/sms/static/src/js/fields_phone_widget.js]: Don't keep commented code, if it is not necessary anymore just remove the line.

File: addons/web/static/tests/fields/basic_fields_mobile_tests.js [hunk 3]
```diff
@@ -111,7 +111,7 @@ QUnit.module('basic_fields', {
 
         assert.containsN(list, '.o_data_row', 3,
             "should have 3 record");
-        assert.strictEqual(list.$('tbody td:not(.o_list_record_selector)').first().text(), 'yop',
+        assert.strictEqual(list.$('tbody td:not(.o_list_record_selector)').first().text().trim(), 'yop',
```
Comment (tde-banana-odoo) [inline addons/web/static/tests/fields/basic_fields_mobile_tests.js]: Why this ?

**Review by tde-banana-odoo — COMMENTED** [review]
Kloug.

**Comment by ryv-odoo** [comment]
@std-odoo , The related task has been cancelled, you can close this PR.

**Pull request closed without merging**